src/pybind/mgr/volumes/fs/operations/versions/metadata_manager.py

```diff
@@ -113,7 +113,7 @@ def flush(self):
 
     def init(self, version, typ, path, state):
         # you may init just once before refresh (helps to overwrite conf)
-        if self.config.has_section(MetadataManager.GLOBAL_SECTION):
+        if self.has_section(MetadataManager.GLOBAL_SECTION):
```

Comment:
If reducing typing is the goal, better to use decorators, isn't it?

Reply:
A helper method suffices IMO.

Reply:
It's really not about sufficing - even the way its currently written suffices.

Could you explain why helper would be better than decorator in this case? (esp. when decorators can throw exceptions too).

Reply:
@rishabh-d-dave ping?

Reply:
I haven't re-visited the initial commits, I've planned to do so after progress bar stuff is finished.

Reply:
ping?

Reply:
I'll get it done once `rstats` are incorpoated and `traverse_tree()` is thrown out.

Reply:
Yes.

Reply:
Done.